pkg/solana/client/multinode/node_lifecycle.go

```diff
@@ -365,13 +368,14 @@ func (n *node[CHAIN_ID, HEAD, RPC]) isOutOfSyncWithPool(localState ChainInfo) (o
 	}
 	// Check against best node
 	ln, ci := n.poolInfoProvider.LatestChainInfo()
+	_, localChainInfo := n.StateAndLatest()
```

Comment:
```suggestion
	localChainInfo, _ := n.rpc.GetInterceptedChainInfo()
```
`StateAndLatest` recalculates Node's state on every call. It's preferred to use `GetInterceptedChainInfo`